src/test/java/hu/dungeonhunter/TestFight.java

```diff
@@ -12,408 +12,499 @@
 
 public class TestFight {
 
-    private MonsterFactory monsterFactory = new MonsterFactory();
     private SoftAssertions softly = new SoftAssertions(); // akkor van értelme amikor több assert van egy metódusban,
-    // az összes hibásat kiadja, nem csak az elsőt.
+    // az összes hibásat kiadja, nem csak az elsőt. softly.assertAll(); //ezzel gyűjti be a hibás asserteket, ha ez nincs, akkor nem dobja fel a hibákat!!!
 
-    @Test
-    public void championAttackTest() {
-        // test setup (Given)
-        Fight fight = new Fight();
-        MonstersInterface lowHpmonster = monsterFactory.getMonster(CharacterTypes.GOBLIN, 1);
-        fight.setMonster(lowHpmonster);
+    private MonsterFactory monsterFactory = new MonsterFactory();
+    private Fight fight = new Fight();
+    private Champion champion = new Champion();
 
-        // test action (When)
-        fight.championAttack();
+    //segítő változók
+    int toTestMonsterCounter = fight.getMonsterCounter();
+    int toTestKilledMonsterCounter = monsterFactory.getKilledMonsterCounter();
+    int toTestHealingPotionCounter = champion.getHealingPotionCounter();
```

Comment:
Please rename these to indicate these are the initialized values. e.g.: startingMonsterCounter

Reply:
I did it, ty!